src/component.mojo

```diff
@@ -116,84 +127,55 @@ struct ComponentManager:
     alias Id = SIMD[Self.dType, 1]
-
     alias max_size = get_max_uint_size[Self.Id]()
-    var _components: Dict[Int, ComponentInfo]
 
     fn __init__(inout self):
         constrained[
             Self.dType.is_integral(),
             "dType needs to be an integral type.",
         ]()
-        self._components = Dict[Int, ComponentInfo]()
+        constrained[
+            len(VariadicList(component_types)) <= Self.max_size,
+            "At most " + str(Self.max_size) + " component types are allowed.",
+        ]()
 
-    fn _register[
-        T: ComponentType, check_existent: Bool = True
-    ](inout self) raises -> ComponentInfo as component_info:
-        """Register a new component type.
+    @always_inline
+    fn get_id[T: ComponentType](self) -> Self.Id:
+        """Get the ID of a component type.
 
         Parameters:
-            T: The component type to register.
-            check_existent: Whether to check if the component type has already been registered.
-
-        Raises:
-            Error: If check_existent and the component type has already been registered.
-            Error: If the maximum number of components has been reached.
+            T: The component type.
         """
 
         @parameter
-        if check_existent:
-            if T.get_type_identifier() in self._components:
-                raise Error(
-                    "A component with hash "
-                    + str(T.get_type_identifier())
-                    + " has already been registered."
-                )
-
-        if len(self._components) >= self.max_size:
-            raise Error(
-                "We cannot register more than "
-                + str(self.max_size)
-                + " elements in a component manager of this type."
-            )
-
-        component_info = ComponentInfo.new[T](Self.Id(len(self._components)))
-        self._components[T.get_type_identifier()] = component_info
-
-    fn get_id[T: ComponentType](inout self) raises -> Self.Id:
-        """Get the ID of a component type.
+        for i in range(len(VariadicList(component_types))):
 
-        If the component does not yet have an ID, register the component.
+            @parameter
+            if _type_is_eq[T, component_types[i]]():
+                return i
 
-        Parameters:
-            T: The component type.
+        # This constraint will fail if the component type is not in the list.
+        constrained[
+            _contains_type[T, *component_types](),
+            "The used component is not in the component parameter list.",
+        ]()
 
-        Raises:
-            Error: If the component was not registered and the maximum number of components has been reached.
-        """
-        if T.get_type_identifier() in self._components:
-            return self._components[T.get_type_identifier()].id
-        else:
-            return self._register[T, False]().id
+        # This is unreachable.
+        return -1
```

Comment:
Maybe raise an error here?

Reply:
Then I would need to make the function raising. For an unreachable error, this would be a disproportionate cost.